src/extension-support/extension-manager.js

```diff
@@ -171,16 +155,33 @@ class ExtensionManager {
      * @returns {Promise} resolved once all the extensions have been reinitialized
      */
     refreshBlocks () {
-        const allPromises = Array.from(this._loadedExtensions.values()).map(serviceName =>
-            dispatch.call(serviceName, 'getInfo')
+        const loadedExtensions = this.runtime.getLoadedExtensions();
+        const allPromises = [];
+        loadedExtensions.forEach((extensionInfo, oldId) => {
+            const serviceName = extensionInfo.serviceName;
+            allPromises.push(dispatch.call(serviceName, 'getInfo')
```

Comment:
I think the `dispatch.call` here should be on its own line for a little bit more clarity when reading.